File: Day1/c++/src/main.cpp
#include "functions.h"
#include <fstream>
#include <iostream>
#include <string>

int main()
{
	int sum=0;
	std::ifstream file("input.txt");
	
	if (!file.is_open()) {
        	std::cerr << "Error opening file" << std::endl;
        	return 1;
    	}

	std::string line;
	
	while(std::getline(file, line)){
	int value= findDigits(line);
	std::cout<<std::endl<<value;
	sum += value;
	}
	
	file.close();
	std::cout<<std::endl<<sum<<std::endl;
	return 0;
}

File: Day1/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main(){
	sum :=0
	//open file
	file, err := os.Open("input.txt")
	if err != nil {
		fmt.Fprintln(os.Stderr, "Error opening file:", err)
	os.Exit(1)
	}

	defer file.Close()


	//read line from file
	scanner := bufio.NewScanner(file)
	for scanner.Scan(){
		line := scanner.Text()
		value := findDigits(line)
		sum += value
	}
	//output result
	fmt.Println(sum)
}
